### PR TITLE
VPN-4496 - Record health metrics on the Android daemon

### PR DESCRIPTION
**Pull request by brizental**
The connection health checks on the Daemon are different from the connection health checks on the client. This is a caveat worth noting at analysis time. I have added a longer description to the metrics to make this more explicit.

For this work I have considered that an unstable event happens when the connection is deemed worh a silent server switch while a no signal even happens when we have given up hope. Does that make sense?

### CODE REVIEW

File: android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/ConnectionHealth.kt [hunk 1]
```diff
@@ -164,8 +166,8 @@ class ConnectionHealth(service: VPNService) {
     private val TaskCheckConnection = Runnable {
         kotlin.run {
             // capture stuff, as this the members will be set by another thread
-            val gateway = mGateway; // This lives inside the tunnel
-            val endpoint = mEndPoint; // This is the wg-endpoint (not in the tunnel)
+            val gateway = mGateway // This lives inside the tunnel
+            val endpoint = mEndPoint // This is the wg-endpoint (not in the tunnel)
```
Comment (brizental) [inline android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/ConnectionHealth.kt:170]: This is not related. I just fixed it while I was fixing other lints.

**Review by strseb — APPROVED** [review]
giphy
